### ISSUE TITLE
Assorted Suggestions #2

### ISSUE THREAD

**Issue by darklordabc** [LegendsOfDota/LegendsOfDota#254]
![image](https://cloud.githubusercontent.com/assets/16277198/14943230/5b307c1c-1016-11e6-922e-b8d5567b1f6d.png)
**Side/Secret Shops don't auto-combine items in Universal shop mode**: Normally an item would combine if it is in my inventory, and in my stash and I am within distance. But it does not do it automatically with the other shops when universal shops are enabled. Also, when you collect items that would normally autocombine, they don't. You have to drop the item on the ground, and then pick it up to proc the auto-combine function. Quite tedious. 

![image](https://cloud.githubusercontent.com/assets/16277198/14943263/3486f216-1017-11e6-8b24-990c721762e4.png)
<del>**Alchemist agh upgrade (Chemical rage) is effectively non-existent:** Alchemist is meant to be able to cast his aghs on another hero, giving them stats and aghs upgrades. However, in LoD the aghs does not become a castable item like it should, even when the game is set to only 1 ultimate each. Clicking it has no effect, hence you cant cast it, hence alchemist aghanim upgrade is broken. </del> [This has it's own topic]

![image](https://cloud.githubusercontent.com/assets/16277198/14943290/dc3e0238-1017-11e6-9456-1d1982bd1736.png)
**Update list of troll combos for Divided We Stand (Meepo ultimate):** Its rather laughable that call down is considered a troll combo, when there is way worst combos that you can pull off. I think you should either remove call down, or add many OP combos. The following are abilities that combine with DWS that are equally bad as call down or worst:
Familiars - You have like 20 birds that insta kill heros, and demolish towers. 
Mass Serpent Wards 
Chaotic Offering
Black Hole
Ravage
Rearm
Exocerism

![image](https://cloud.githubusercontent.com/assets/16277198/14943346/9d4bbf8c-1019-11e6-9a95-ddd26e62061f.png)
<del>**Purifying Flames causing problems since 6.87:** There is a new aghs upgrade for this ability, which radically decreases its cooldown to 1 second. As with all abilities with low cooldowns, this creates some powerful combos. Firstly, it triggers aftershock, and its not considered a troll combo yet. Second, it also triggers essence aura, so its an easy combo for unlimited mana. 
Just an idea to future proof: Procedurally make any ability with 5 second cooldown count as a troll combo with aftershock. The game should check its default cooldown and its aghs cooldown, as it seems more common for aghs to effect level 1 abilities. P.S: I think you may of missed the neutral ability of purge as counting as a troll combo with aftershock, it only has a 3 second cooldown. </del>

![image](https://cloud.githubusercontent.com/assets/16277198/14943367/9290165a-101a-11e6-9592-4dd6750ae60c.png)
<del>**Starting with a courier:** I think it would be a good idea to have a courier spawn automatically for teams. This type of game-mode is not inducive to support players, seeing how nearly everybody tries to build right-click OP heroes. Unfortunately its pretty common to play games where nobody buys a courier at the start and nobody buys wards (maybe like 20% of games). Please either make a courier spawn automatically when game starts (like in 1 vs 1 dota), or add an option under "Game speed" or "Other" for "Start with courier" checkbox. I'm positive that if that option was there, it would always be ticked by players. </del>

**Comment by ash47** [comment]
The meepo one is not really much of an issue. The way we have done banning for the Meepo ult is that global abilities that can be cast anywhere have been blocked. Meepo ult is considered to be OP, people who play with OP abilities enabled generally don't use the troll combos list.

@Jexah

**Comment by ash47** [comment]
We used to have an option that would give a free courier, the code for it is still in LoD actually, trouble is that for what ever reason, the courier that spawns can not be controlled by players, or the hot keys don't work, or something dodgy like that.

**Comment by darklordabc** [comment]
Hey, you are making all these commits, but not updating the game? So I'm thinking, how are you testing that these commits are actually working? Secondly, can I have access to a test client, if it exists, so I can help debug and test?

**Comment by ash47** [comment]
You think I just randomly commit things? I test every piece of code that
goes into the repo.

There is instructions on how to compile lod in the repo readme

https://github.com/ash47/LegendsOfDota/blob/master/README.md

Once compiled, it will create "Dota/content" and "Dota/game", these
corospond to the folders in Dota, you need to create a new add-on and put
the compiled files in, I use directory junctions.

**Comment by darklordabc** [comment]
Hmm have you compiled recently? I run into problems. 
![image](https://cloud.githubusercontent.com/assets/16277198/14990302/53ca1c64-119f-11e6-8369-bba453450074.png)

**Comment by darklordabc** [comment]
Btw still having troubles compiling.

**Comment by darklordabc** [comment]
So after compiling you only need to copy Dota/content and dota/game over to their corresponding folders in the steam folders? Because I've done that, and its still not working.

**Comment by ash47** [comment]
You need to create an add-on and put them into the addon's folder.

**Comment by darklordabc** [comment]
I've done that. But I think something is going wrong in the compiling process. A couple of the mklinks dont work.

**Comment by darklordabc** [comment]
For example "mklink /D /J "dota\content\maps" "src\maps"" But there is no folder maps in the "src" folder? So when I try to open "dota\content\maps" it gives an error of location not found.

**Comment by darklordabc** [comment]
This is as far as I can get, but the GUI to pick options never comes up. 
![image](https://cloud.githubusercontent.com/assets/16277198/15003122/d41d8ab2-11eb-11e6-9969-a2365d79d213.png)

**Comment by ash47** [comment]
You can just make that directory, that SRC maps folder is where I upload
maps when I need to compile, I never commit them to the repo.

Any lua errors in the console?

**Comment by ash47** [comment]
Note that there is two bat files, one compiles the language and other
files, the other stages the files for dota

**Comment by darklordabc** [comment]
I know about the two bat files. I've tried a dozen times so far. 
Only errors in the console are
"[ W General                ]: ILocalize::AddFile() failed to load file "resource/addon_english.txt"."
"[ W General                ]: Script failed to LoadKeyValues scripts/npc/npc_heroes_custom.txt"
I dont think there are any LUA errors.

Have you tried compiling from scratch as if you just downloaded the zip?

**Comment by darklordabc** [comment]
Heres is what I do
1. Unzip the repository.  
2. Make a settings.json file with the right dota 2 directory.
3. Open CMD.exe with admin privileges. Navigate to folder,  and run stage.bat. It produces this [log](http://pastebin.com/kxYTCjda). It produces a few errors. E.g. A few of them reference folders/files that dont exist. 
4. I then run compile.bat. It throws a [few errors](http://pastebin.com/gzk9z3Mh), but mostly seems to work.

**Comment by darklordabc** [comment]
Heres is what I do
1. Unzip the repository.

2. Make a settings.json file with the right dota 2 directory.
3. Open CMD.exe with admin privileges. Navigate to folder, and run stage.bat. It produces this log. It produces a few errors. E.g. A few of them reference folders/files that dont exist. 
4. I then run compile.bat. It throws a [few errors](Heres is what I do
1. Unzip the repository.

2. Make a settings.json file with the right dota 2 directory.
3. Open CMD.exe with admin privileges. Navigate to folder, and run stage.bat. It produces this log. It produces a few errors. E.g. A few of them reference folders/files that dont exist. 
4. I then run compile.bat. It throws a few errors, but mostly seems to work.), but mostly seems to work.

**Comment by darklordabc** [comment]
Heres is what I do
1. Unzip the repository.
2. Make a settings.json file with the right dota 2 directory.
3. Open CMD.exe with admin privileges. Navigate to folder, and run stage.bat. It produces this [log](http://pastebin.com/kxYTCjda). It produces a few errors. E.g. A few of them reference folders/files that dont exist. 
4. I then run compile.bat. It throws some [errors ](http://pastebin.com/gzk9z3Mh) but seems to work.
5. Copy over the files manually from the 2 folders (content and game), over to where they are on steam directly. e.g. (\game\dota_addons\lod_beta and \content\dota_addons\lod_beta). 
6. Start game, launch custom mode with custom map. 
7. Mod does'nt work, appears like its missing files (which I suspect is the case because not all necessary files got compiled into the game and content folders.

**Comment by darklordabc** [comment]
I just want to be able to test modifications :cry:

**Comment by ash47** [comment]
Maybe paste the Dota 2 console log, I suspect there are lua errors

**Comment by darklordabc** [comment]
[Console log](http://pastebin.com/xQBDHPQi) From start of dota2.exe to loading into broken lod gamemode.

**Comment by ash47** [comment]
I took two seconds looking at the console and it reads as follows:

> [   VScript              ]: Legends of dota is activating! (v3.1.1)
> [ W General              ]: Script failed to LoadKeyValues scripts/npc/npc_heroes_custom.txt
> [   VScript              ]: FAILURE! You are attempting to run an UNCOMPILED version! Please either compile OR download the latest release from the releases section of github.

**Comment by ash47** [comment]
Not much I can do about the side shops, you can post a bug report here:

http://dev.dota2.com/forumdisplay.php?f=485

I am using the following funciton:

GameRules:SetUseUniversalShopMode(bool enabled)

**Comment by darklordabc** [comment]
Yeah I noticed that in my investigation. The best option I saw mentioned by someone else, is replacing the glyph button with a custom UI button, and then doing a custom fortify ability. More important things to do tho.
